build_tools/sdk.py

```diff
@@ -0,0 +1,198 @@
+        return None
+    # Xcode 12.5.1
+    # Build version 12E507
+    xcode_version_full = run.shell_command('/usr/bin/xcodebuild -version')
+    xcode_version_lines = xcode_version_full.split("\n")
+    xcode_version = xcode_version_lines[0].split()[1].strip()
+    return xcode_version
+
+def get_local_darwin_sdk_path(platform):
+    return run.shell_command('xcrun -f --sdk %s --show-sdk-path' % _convert_darwin_platform(platform)).strip()
+
+def get_local_darwin_sdk_version(platform):
+    return run.shell_command('xcrun -f --sdk %s --show-sdk-platform-version' % _convert_darwin_platform(platform)).strip()
+
+
+## **********************************************************************************************
+
+# ANDROID_HOME
+
+def _get_defold_path(sdkfolder, platform):
+    return os.path.join(sdkfolder, defold_info[platform]['pattern'])
+
+def check_defold_sdk(sdkfolder, platform):
+    folders = []
+
+    if platform in ('x86_64-darwin', 'armv7-darwin', 'arm64-darwin', 'x86_64-ios'):
+        folders.append(_get_defold_path(sdkfolder, 'xcode'))
+        folders.append(_get_defold_path(sdkfolder, platform))
+
+    if platform in ('x86_64-win32', 'win32'):
+        folders.append(os.path.join(sdkfolder, 'Win32','WindowsKits','10'))
+        folders.append(os.path.join(sdkfolder, 'Win32','MicrosoftVisualStudio14.0','VC'))
+    if platform in ('armv7-android', 'arm64-android'):
+        folders.append(os.path.join(sdkfolder, "android-ndk-r%s" % ANDROID_NDK_VERSION))
+        folders.append(os.path.join(sdkfolder, "android-sdk"))
+
+    ok = True
+    for f in folders:
+        if not os.path.exists(f):
+            log("Missing SDK in %s" % f)
+            ok = False
+    return ok
+
+
+def check_local_sdk(platform):
+    if platform in ('x86_64-darwin', 'armv7-darwin', 'arm64-darwin', 'x86_64-ios'):
+        xcode_version = get_local_darwin_toolchain_version()
+        if not xcode_version:
+            return False
+    return True
+
+
+def _get_defold_sdk_info(sdkfolder, platform):
+    info = {}
+    if platform in ('x86_64-darwin','x86_64-ios','armv7-darwin','arm64-darwin'):
+        info['xcode'] = {}
+        info['xcode']['version'] = VERSION_XCODE
+        info['xcode']['path'] = _get_defold_path(sdkfolder, 'xcode')
+        info['xcode-clang'] = defold_info['xcode-clang']
+        info[platform] = {}
+        info[platform]['version'] = defold_info[platform]['version']
+        info[platform]['path'] = _get_defold_path(sdkfolder, 'xcode')
+
+    return info
+
+def _get_local_sdk_info(platform):
+    info = {}
+    if platform in ('x86_64-darwin','x86_64-ios','armv7-darwin','arm64-darwin'):
+        info['xcode'] = {}
+        info['xcode']['version'] = get_local_darwin_toolchain_version()
+        info['xcode']['path'] = get_local_darwin_toolchain_path()
+        info[platform] = {}
+        info[platform]['version'] = get_local_darwin_sdk_version(platform)
+        info[platform]['path'] = get_local_darwin_sdk_path(platform)
```

Comment:
Each platform has its own requirements, so we allow it to output whatever is needed by the build.py/waf_dynamo.py